### PR TITLE
fix: Form remote content type

### PR DESCRIPTION
**Pull request by jefersonlopeszup**
### Description

BeagleApi included in FormSubmitter and simplified CoroutineScope 

## Related Issues

#410

## Tests

I change the following tests:

FormSubmitterTest
HttpClientDefaultTest

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [DCO].
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/09c3c63e5c5dbe74f45c50b9ec23160f0aac4328/android/sample/src/main/res/values/styles.xml#L116">android/sample/src/main/res/values/styles.xml#L116</a> - The resource <code>R.style.DesignSystem_Text_NetworkImage</code> appears to be unused</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | Classname | Time|
--- | --- | ---|
execute_should_set_HttpMethod_PATCH | br.com.zup.beagle.android.networking.HttpClientDefaultTest | 0.172|
execute_should_set_headers | br.com.zup.beagle.android.networking.HttpClientDefaultTest | 0.149|
execute_should_disconnect_after_response | br.com.zup.beagle.android.networking.HttpClientDefaultTest | 0.165|


<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/android/utils/CoroutineScopeBase.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+package br.com.zup.beagle.android.utils
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlin.coroutines.CoroutineContext
+
+abstract class CoroutineScopeBase : CoroutineScope {
```
Comment (uziasferreirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/utils/CoroutineScopeBase.kt:24]: Be careful when using base classes.
In this case I don't think it's a good approach to create a base class and use it in `HttpClientDefault`, `BeagleServiceWrapper`, there is no concrete relationship.
Why don't you add an attribute to your classes called `scope` and change this class to `DefaultAndroidCoroutineScope`?

Example:
```
abstract class DefaultAndroidCoroutineScope : CoroutineScope {
......
}

internal class FormSubmitter(
    private val httpClient: HttpClient = HttpClientFactory().make(),
    private val beagleApi: BeagleApi = BeagleApi(),
    private val deserialization: BeagleSerializer = BeagleSerializer(),
    private val urlBuilder: UrlBuilder = UrlBuilderFactory().make(),
   private val scope: DefaultAndroidCoroutineScope = DefaultAndroidCoroutineScope()
) {
.......
}

```

Comment (jefersonlopeszup) [reply android/beagle/src/main/java/br/com/zup/beagle/android/utils/CoroutineScopeBase.kt]: Thanks for you feed back.

**Pull request merged**